docs/contributor-guide/README.md

```diff
@@ -0,0 +1,109 @@
+# KubeDB Operator Design Document
```

Comment:
I would prefer some `Installation` and `testing` guide first before jumping into code itself.